### ISSUE TITLE
No filters present on file selection dialogs

### ISSUE THREAD

**Issue by aerwin** [maqetta/maqetta#1725]
I ran across this while working on #1638... It appears that none of the dialogs that present a file chooser tree offer extension based filtering of the choices. E.g., while only `*.json` files make sense for `DataGridInput`, files of all types can be selected. 

In all cases where a tree definition is passed to `davinci.ui.Panel`, the `filters` parm is specified as follows:

``` javascript
filters: "new system.resource.FileTypeFilter(parms.fileTypes || '*');"
```

But, the `davinci.ui.Panel`code code attempts to do a `dojo.getObject` on the specified filter(s) so this would never get properly evaluated..

(Furthermore, I don't ever see `parms.fileTypes` set anywhere, so even if `dojo.getObject` could handle this, the filter will always be the single wild card anyway.)

I figured as part of re-factoring for #1638, I could just fix this by manually creating instances of `system.resource.FileTypeFilter` and specifying better filter strings. But, `system.resource.FileTypeFilter` is in the orion project and I was having trouble loading it. So, figured I could deal with this as a separate issue after #1638 is addressed (since filtering isn't doing anything now anyway).

**Comment by JonFerraiolo** [comment]
Sounds like a good plan to me. We do want filter functionality in the future. IMO, basically Maqetta and Orion need to reproduce all of the same sorts of UI niceties in our file dialogs and palettes as we have in desktop UIs.

@childsb, @peller

**Comment by peller** [comment]
It should be possible to filter the items in the tree using the dojo.data layer, and system.resource follows the dojo.data API.  Some of our old code used to mess with dijit.Tree and filter/sort directly in the UI, which was problematic.

**Comment by aerwin** [comment]
Given this hasn't worked recently (sounds like from @peller some code that maybe used to kind of work was removed awhile ago) and the long-term desire stated by @JonFerraiolo is to make this file chooser on par with a desktop UI file chooser, I'm going to reclassify this as a Feature Request. In addition, I don't think this makes sense for M5 given scope and other priorities. So, for now I'm moving to "future". @JonFerraiolo or @peller if you have different opinions on milestone to target this feature for, please let me know.
